include/experimental/__p0009_bits/mdspan.hpp

```diff
@@ -376,6 +442,23 @@ class mdspan
   MDSPAN_FORCE_INLINE_FUNCTION constexpr mapping_type const& __mapping_ref() const noexcept { return __members.__second().__first(); }
   MDSPAN_FORCE_INLINE_FUNCTION _MDSPAN_CONSTEXPR_14 accessor_type& __accessor_ref() noexcept { return __members.__second().__second(); }
   MDSPAN_FORCE_INLINE_FUNCTION constexpr accessor_type const& __accessor_ref() const noexcept { return __members.__second().__second(); }
+  
+  MDSPAN_TEMPLATE_REQUIRES(
+    class SizeType,
+    /* requires */ (
+      _MDSPAN_TRAIT(std::is_convertible, const SizeType&, index_type) &&
+      _MDSPAN_TRAIT(std::is_nothrow_constructible, index_type, const SizeType&)
+    )
+  )
+  MDSPAN_FORCE_INLINE_FUNCTION constexpr bool __is_index_oor(SizeType index, index_type extent) const noexcept {
+    // Check for negative indices
+    if constexpr(std::is_signed_v<SizeType>) {
+      if(index < 0) {
+        return true;
+      }
+    }
```

Comment:
`SizeType` doesn't need to be integral here or even less-than comparable with `int`.  mdspan behaves as if all operations on indices happen after conversion to `index_type`.  Thus, should we consider instead converting to `index_type` first, as we do below?

```suggestion
    // Check for negative indices
    if constexpr (std::is_signed_v<index_type>) {
      if (static_cast<index_type>(index) < 0) {
        return true;
      }
    }
```

Reply:
Seems reasonable

Reply:
After thinking about it again, I'm not sure if this is correct: imagine I have `unsigned short` as `index_type` and an extent of size of `65535`. Now, I want to access with a `signed short` of value `-10`. What I end up getting is the element at index `65526`, which is not the expected result.